pyteal/ast/subroutine.py

```diff
@@ -1,18 +1,25 @@
 from typing import Callable, List, Optional, TYPE_CHECKING
-from inspect import Parameter, signature
+from inspect import Parameter, isclass, signature
 
 from ..types import TealType
 from ..ir import TealOp, Op, TealBlock
 from ..errors import TealInputError, verifyTealVersion
 from .expr import Expr
 from .seq import Seq
 from .scratchvar import ScratchVar
+from . import abi
 
 if TYPE_CHECKING:
     from ..compiler import CompileOptions
 
+# NOTE trying to do return, rather than "write to reference", experimenting here
 
 class SubroutineDefinition:
+    PARAM_ANNOTATION_TYPES = (Expr, ScratchVar, abi.Type)
```

Comment:
kewl

Reply:
nope, this seem to be wrong, let me explain: the check sees if an annotation type is **exactly** in `PARAM_ANNOTATION_TYPES`, but for `abi.Type`, what we are looking for is its sub-types.
Some changes need to be done here...